scripts/prepare-beta.ts

```diff
@@ -0,0 +1,97 @@
+import * as execa from 'execa';
+import { readFileSync, readdirSync, writeFileSync, statSync } from 'fs';
+import * as path from 'path';
+
+let lernaModules: string[] = [];
+const SCOPE = '@bitgo-beta';
+
+/**
+ * Create a function which can run lerna commands
+ * @param {String} lernaPath - path to lerna binary
+ * @returns {function(string, string[], Object.<string, string>): Promise<string>}
+ */
+function getLernaRunner(lernaPath: string) {
+  return async (command: string, args: string[] = [], options = {}) => {
+    const { stdout } = await execa(
+      lernaPath,
+      [command, ...args],
+      options,
+    );
+    return stdout;
+  };
+}
+
+const getLernaModules = async (): Promise<void> => {
+  const { stdout: lernaBinary } = await execa('yarn', ['bin', 'lerna'], { cwd: process.cwd() });
+
+  const lerna = getLernaRunner(lernaBinary);
+  const modules: Array<{name: string}> = JSON.parse(await lerna('list', ['--loglevel', 'silent', '--json', '--all']));
+  lernaModules = modules.map(({ name }) => name);
+};
+
+const walk = (dir: string): string[] => {
+  let results: string[] = [];
+  const ignoredFolders = [/modules\/bitgo\/example/, /node_modules/];
+  try {
+    const list = readdirSync(dir);
+    list.forEach((file) => {
+      file = path.join(dir, file);
+      const stat = statSync(file);
+      if (stat && stat.isDirectory()) {
+        if (!ignoredFolders.some((folder) => folder.test(file))) {
+          results = [
+            ...results,
+            ...walk(file),
+          ];
+        }
+      } else if (['.ts', '.tsx', '.js', '.json'].includes(path.extname(file))) {
+        // Is a file
+        results.push(file);
+      }
+    });
+  } catch (error) {
+    console.error(`Error when walking dir ${dir}`, error);
```

Comment:
@bitgoAaron is it safe to fail silently here? Since the releases will be automated, I feel like we should abort, unless errors are expected when walking the directories.

Reply:
yeah, good call. I'll still log the error so we know which directory/file was problematic and throw the exception instead.